integration-tests/src/test/java/tachyon/worker/BlockServiceHandlerIntegrationTest.java

```diff
@@ -280,9 +297,13 @@ private void createBlockFile(String filename, int len) throws IOException, Inval
     out.close();
   }
 
-  // Sleeps for a duration so that the worker heartbeat to master can be processed
-  private void waitForHeartbeat() {
-    CommonUtils
-        .sleepMs(mWorkerTachyonConf.getInt(Constants.WORKER_BLOCK_HEARTBEAT_INTERVAL_MS) * 3);
+  // Waits for a worker heartbeat to master to be processed
+  private void waitForHeartbeat() throws InterruptedException {
```

Comment:
This is okay if we only ever wait for a single heartbeat. However, calling this method twice would result in a hang.

It might make more sense to add the initial `await` call into the `before` method an have this method only invoke `schedule` followed by `await`.